pkg/tbtc/tbtc.go

```diff
@@ -56,26 +56,29 @@ func Initialize(
 	workPersistence persistence.BasicHandle,
 	scheduler *generator.Scheduler,
 	config Config,
-	registry *diagnostics.Registry,
+	metricsRegistry *metrics.Registry,
 ) error {
 	node := newNode(chain, netProvider, keyStorePersistence, workPersistence, scheduler, config)
 	deduplicator := newDeduplicator()
 
-	registry.RegisterApplicationSource(
-		"tbtc",
-		func() map[string]interface{} {
-			return map[string]interface{}{
-				"preParamsPoolSize": node.dkgExecutor.PreParamsCount(),
-			}
-		},
-	)
+	if metricsRegistry != nil {
+		// only if metrics are configured
+		metricsRegistry.ObserveApplicationSource(
+			"tbtc",
+			map[string]metrics.Source{
+				"pre_params_in_pool": func() float64 {
```

Comment:
How about simplifying the name to match the function it calls?
```suggestion
				"pre_params_count": func() float64 {
```

Reply:
Sure!

Reply:
836bec984adbbecefbbfe91f7a3fd446169d20b8